.gitlab-ci.yml

```diff
@@ -1,14 +1,108 @@
-test:
-  image: "python:3.6-stretch"
-  script:
-    - apt-get update -qq
-    - apt-get install -y git
-    - apt-get install -y xsltproc
-    - apt-get install -y genisoimage
-    - apt-get install -y enchant
-    - apt-get install -y shellcheck
-    - apt-get install -y python3-pip
-    - pip3 install --upgrade pip
-    - pip install tox
+stages:
+  - test
+  - doc
+  - package
+  - deploy
+
+.fedora_install_deps: &fedora_install_deps
+  before_script:
+    - dnf install -y --refresh python3 python3-devel python3-pip
+    # requirements
+    - dnf install -y python3-docopt python3-lxml python3-pyyaml python3-pyxattr python3-six python3-future python3-requests
+    # dev-requirements
+    - dnf install -y python3-pytest python3-pytest-xdist python3-pytest-cov python3-mock python3-flake8 python3-sphinx python3-sphinx_rtd_theme.noarch python3-sphinxcontrib-spelling python3-enchant python3-tox make gcc which genisoimage xz
```

Comment:
We're generally using `xorriso` instead of `genisoimage` for kiwi.

Reply:
I just tried that. The tests fail when genisoimage is not installed: https://gitlab.com/D4N/kiwi-ci/-/jobs/171908894.

Is that expected?

Reply:
No. That seems broken... @schaefi?

Reply:
I fixed that part in #c4f0855d1f160070982b0609d2cea831751cdc47

Reply:
That's really odd, since that commit is included in this PR.

Reply:
Hmm, I see I didn't catch this case. So there are two unit tests that actually test the content list of an iso for the cdrtools case. This tests read the example iso file using isoinfo and expects the tool to be present. I think we need a @pytest.mark.skipif there is no genisoimage installed. So that needs fixing

Reply:
https://github.com/SUSE/kiwi/pull/982

that should make it work

Reply:
I'll rebase my branch on master once #982 get's merged.

Reply:
It's been merged now, so this should be rebased and fixed :)

Reply:
*grmbl* I take a look

Reply:
https://github.com/SUSE/kiwi/pull/990

This should finally make the tests happy